### ISSUE TITLE
Duplicate data

### ISSUE THREAD

**Issue by Rockforced** [angular/angularfire2#987]
On some lists, whenever you subscribe or whenever you change any child data on said list, it returns a list which has duplicates of each child on the list. I've seen other people have this issue, I really can't release my ionic app until this is fixed. At least on ionic, if you update the list on a completely different component, and then navigate back to where you .map or .subscribe to the list, the list either contains more duplicates or has fixed itself. Reloading the component or refreshing the app produces the same duplicates. Please look into this.

### Version info

**Angular:**
4.0.0
**Firebase:**
3.9.0
**AngularFire:**
2.0.0-beta.7-pre
**Ionic 3**

### Steps to reproduce

https://youtu.be/8fwJNTMpx6U

Subscribe to a list, then update any child on the list. In my case, the map duplicates each item.

**Comment by Rockforced** [comment]
I added a video that describes the situation.

I found out that it appears to happen only the initial subscription to the firebase location.
So if I have,
  ` 
      this.units = this.entityService.getUnitList(this.navParams.data.property_id);
`
then I end up with duplicate data. However if I wrap that subscription as such:
`    this.entityService.getUnitList(this.navParams.data.property_id).subscribe(_units => {
      this.units = this.entityService.getUnitList(this.navParams.data.property_id);
    });` then I end up with no duplicate data.

What is going on?
